src/components/Common/CopyButton.tsx

```diff
@@ -0,0 +1,58 @@
+import { useState } from "react";
+
+import CareIcon, { IconName } from "@/CAREUI/icons/CareIcon";
+
+import { TooltipComponent, TooltipProvider } from "@/components/ui/tooltip";
+
+import { copyToClipboard } from "@/Utils/utils";
+
+interface CopyButtonProps {
+  content: string;
+  tooltipContent?: string;
+  btnContent?: string;
+  resetDuration?: number;
+  iconClassName?: string;
+  btnClassName?: string;
+  icons?: {
+    copied: IconName;
+    copy: IconName;
+  };
+}
```

Comment:
Looks scary. Let's keep it super simple.

```suggestion
interface CopyButtonProps {
  content: string;
}
```

Reply:
we might still need btnContent ans btnClassName, for SBOM viewer

also some copy icons are bigger than others , so we might need iconClassName for them

are you sure we don't need custom tooltip content? @rithviknishad

Reply:
let's use CVA for the size variant for achieving sizes. that should eliminate the need for btnClassName and iconClassName.

btnContent can be renamed to `children` instead.